geoapps/inversion/line_sweep/driver.py

```diff
@@ -87,15 +90,55 @@ def collect_results(self):
         )
 
         data = {}
+        drape_models = []
         for line in lines:
-            ws = Workspace(f"{os.path.join(path, files[line])}.ui.geoh5")
-            survey = ws.get_entity("Data")[0]
-            data = self.collect_line_data(survey, data)
-            mesh = ws.get_entity("Models")[0]
-            mesh = mesh.copy(parent=models_group)
-            mesh.name = f"Line {line}"
+            with Workspace(f"{os.path.join(path, files[line])}.ui.geoh5") as ws:
+                survey = ws.get_entity("Data")[0]
+                data = self.collect_line_data(survey, data)
+                mesh = ws.get_entity("Models")[0]
+                mesh = mesh.copy(parent=models_group)
+                mesh.name = f"Line {line}"
+                drape_models.append(mesh)
 
         data_result.add_data(data)
```

Comment:
Hmmm, this would have to be inside the context to work...

Reply:
Don't think so.  The above context is for reading the line workspaces, so I can pile up results in the data which is then added to data_result whose workspace is self.workspace (still open in the context started at line33)

Reply:
Got it, different workspace.